api/_examples/pagination/main.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/lacework/go-sdk/api"
+)
+
+func main() {
+	lacework, err := api.NewClient(os.Getenv("LW_ACCOUNT"), api.WithApiV2(),
+		api.WithApiKeys(os.Getenv("LW_API_KEY"), os.Getenv("LW_API_SECRET")))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	response, err := lacework.V2.Entities.ListMachineDetails()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for {
+		// Output: Rows read: 5000
+		// Output: Rows read: 5000
+		// Output: Rows read: 21
+		fmt.Printf("Rows read: %d\n", response.Paging.Rows)
+
+		pageOk, err := lacework.NextPage(&response)
+		if err == nil && pageOk {
+			continue
+		}
+		break
+	}
+}
```

Comment:
Following on from our zoom I think i'd prefer an approach where we can call NextPage on the response.

```
	response, err := lacework.V2.Entities.ListMachineDetails()
	if err != nil {
		log.Fatal(err)
	}

	rowCount := response.PageInfo().Rows

	for {
		fmt.Printf("showing %v rows out of %v \n", rowCount, response.PageInfo().TotalRows)
		if response.HasNextPage() {
			continue
		}
		response.GetNextPage(&response)
		rowCount = rowCount + response.PageInfo().Rows
		break
	}
```

Reply:
To do this, **every single response** would have to have a copy of the client (that is `*api.Client`).

Are we ok with that?

Reply:
No I don't think that would be a good idea. Could we shift a NextPage implementation to the entities services?

Reply:
I thought about it too, why do you want that move? 

If we do that, then we won't have logging at the service level because `Client.log` is private. Either we make it public or we don't have logging. (we can definitely move it, there are just these tradeoffs)